### ISSUE TITLE
Compilation failure when building mpv

### ISSUE THREAD

**Issue by donradzsk** [m-ab-s/media-autobuild_suite#1526]
First window:
```
-------------------------------------------------------------------------------
Updating pacman database...
-------------------------------------------------------------------------------

:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date

-------------------------------------------------------------------------------
Checking pacman packages...
-------------------------------------------------------------------------------


-------------------------------------------------------------------------------
You're missing some packages!
Do you want to install them?
-------------------------------------------------------------------------------

Install:
mingw-w64-i686-luajit-git
mingw-w64-x86_64-luajit-git
install packs [y/n]? y
warning: mingw-w64-i686-luajit-2.0.5-1 is up to date -- skipping
warning: mingw-w64-x86_64-luajit-2.0.5-1 is up to date -- skipping
 there is nothing to do
warning: mingw-w64-i686-luajit-2.0.5-1 is up to date -- skipping
warning: mingw-w64-x86_64-luajit-2.0.5-1 is up to date -- skipping
 there is nothing to do
error: could not set install reason for package mingw-w64-i686-luajit-git (could not find or read package)
error: could not set install reason for package mingw-w64-x86_64-luajit-git (could not find or read package)
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
```
Second window:
```
        Starting 32bit compilation of all tools
Imported default FFmpeg options from .bat
License: nonfree
Imported default mpv options from .bat

        Starting 32bit compilation of global tools
├ Running rustup_update...
├ Running set_default_toolchain...
sccache git  ............................................. [Already built]
ripgrep git  ............................................. [Already built]
jo git  .................................................. [Already built]
jq git  .................................................. [Already built]
dssim git  ............................................... [Already built]
libxml2 git  ............................................. [Already built]
freetype2 git  ........................................... [Already built]
fontconfig git  .......................................... [Already built]
harfbuzz git  ............................................ [Already built]
fribidi git  ............................................. [Already built]
libass git  .............................................. [Already built]
gnutls 3.6.10  .............................................. [Up-to-date]
libidn2 2.2.0  .............................................. [Up-to-date]
libpsl 0.21.0  .............................................. [Up-to-date]
curl git  ................................................ [Already built]
libtiff git  ............................................. [Already built]
libwebp git  ............................................. [Already built]
├ FFmpeg and related apps will depend on OpenCL.dll
OpenCL-Headers git  ...................................... [Already built]
leptonica git  ........................................... [Already built]
tesseract git  ........................................... [Already built]
rubberband 1.8.1  ........................................... [Up-to-date]
zimg git  ................................................ [Already built]

        Starting 32bit compilation of audio tools
libilbc 2.0.3-dev  .......................................... [Up-to-date]
flac git  ................................................ [Already built]
vo-amrwbenc 0.1.3  .......................................... [Up-to-date]
fdk-aac git  ............................................. [Already built]
bin-fdk-aac git  ......................................... [Already built]
opus git  ................................................ [Already built]
opusfile git  ............................................ [Already built]
libopusenc git  .......................................... [Already built]
opus-tools git  .......................................... [Already built]
libsoxr git  ............................................. [Already built]
codec2 git  .............................................. [Already built]
lame 3.100  ................................................. [Up-to-date]
libgme 0.6.2  ............................................... [Up-to-date]
libbs2b 3.1.0  .............................................. [Up-to-date]
sndfile git  ............................................. [Already built]
sox 14.4.2  ................................................. [Up-to-date]
openmpt git  ............................................. [Already built]
libmysofa git  ........................................... [Already built]
flite git  ............................................... [Already built]
shine 3.1.1  ................................................ [Up-to-date]
openal-soft git  ......................................... [Already built]

        Starting 32bit compilation of video tools
librtmp git  ............................................. [Already built]
vpx git  ................................................. [Already built]
aom git  ................................................. [Already built]
dav1d git  ............................................... [Already built]
rav1e git  ............................................... [Already built]
kvazaar git  ............................................. [Already built]
sdl2 2.0.10  ................................................ [Up-to-date]
dvdread git  ............................................. [Already built]
dvdnav git  .............................................. [Already built]
libaacs git  ............................................. [Already built]
libbdplus git  ........................................... [Already built]
libbluray git  ........................................... [Already built]
xavs 0.1  ................................................... [Up-to-date]
libzen git  .............................................. [Already built]
libmediainfo git  ........................................ [Already built]
mediainfo git  ........................................... [Already built]
vidstab 1.10  ............................................... [Up-to-date]
zvbi-0.2 0.2.35  ............................................ [Up-to-date]
dlfcn-win32 git  ......................................... [Already built]
frei0r git  .............................................. [Already built]
decklink-headers git  .................................... [Already built]
libmfx git  .............................................. [Already built]
AMF git  ................................................. [Already built]
gpac git  ................................................ [Already built]
x264 git  ................................................ [Already built]
x265 hg  ................................................. [Already built]
ffnvcodec git  ........................................... [Already built]
srt git  ................................................. [Already built]
python 3.7.5  ............................................... [Up-to-date]
vapoursynth R48  ............................................ [Up-to-date]
  Running git update for ffmpeg...
ffmpeg git  ................................................. [Up-to-date]
├ Compiling mpv with Vapoursynth R48
├ mpv will need vapoursynth.dll and vsscript.dll to use vapoursynth filter!
mujs git  ................................................ [Already built]
vulkan-loader git  ....................................... [Already built]
shaderc git  ............................................. [Already built]
SPIRV-Cross git  ......................................... [Already built]
libplacebo git  .......................................... [Already built]
  Running git clone for mpv...
┌ mpv git  ............................................ [Recently updated]
├ Running bootstrap...
├ Uninstalling python3-docutils... failed
├ Running configure...
├ Running build...
Likely error (tail of the failed operation logfile):
                 from C:/MingW/media-autobuild_suite-master/local32/include/libavcodec/avcodec.h:31,
                 from C:/MingW/media-autobuild_suite-master/local32/include/libavformat/avformat.h:319,
                 from ../demux/demux_lavf.c:30:
C:/MingW/media-autobuild_suite-master/local32/include/libavutil/mem.h:342:1: warning: 'alloc_size' attribute ignored on a function returning 'int' [-Wattributes]
  342 | av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
      | ^~~~~~~~~~~~~

Waf: Leaving directory `/build/mpv-git/build'
Build failed
 -> task in 'objects' failed with exit status 1 (run with -v to display more information)
build failed. Check C:/MingW/media-autobuild_suite-master/build/mpv-git/ab-suite.build.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/z78-.zip
Copy and paste [logs.zip](https://0x0.st/z78-.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```
I keep deleting the mpv-git folder in hopes that it will eventually succeed but it always fails at the same point.

**Comment by 1480c1** [comment]
```none
../video/out/gpu/spirv_shaderc.c: In function 'shaderc_init':
../video/out/gpu/spirv_shaderc.c:35:37: error: 'shaderc_optimization_level_performance' undeclared (first use in this function); did you mean 'shaderc_optimization_level_zero'?
   35 |                                     shaderc_optimization_level_performance);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                     shaderc_optimization_level_zero
```

**Comment by donradzsk** [comment]
OK I changed performance to zero and it failed somewhere else
```
In file included from C:/MingW/media-autobuild_suite-master/local32/include/libavutil/common.h:488,
                 from ../stream/stream_dvdnav.c:41:
C:/MingW/media-autobuild_suite-master/local32/include/libavutil/mem.h:342:1: warning: 'alloc_size' attribute ignored on a function returning 'int' [-Wattributes]
  342 | av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
      | ^~~~~~~~~~~~~

Waf: Leaving directory `/build/mpv-git/build'
Build failed
 -> task in 'objects' failed with exit status 1 (run with -v to display more information)
```

**Comment by 1480c1** [comment]
Those are warnings, probably same error. I will look into it when I have time

**Comment by 1480c1** [comment]
Try deleting `shaderc-git` folder under build and rerun

**Comment by donradzsk** [comment]
Looks like the same thing.
```
┌ mpv git  ............................................ [Recently updated]
├ Running bootstrap...
├ Uninstalling python3-docutils... failed
├ Running configure...
├ Running build...
Likely error (tail of the failed operation logfile):
../video/out/gpu/spirv_shaderc.c:35:37: note: each undeclared identifier is reported only once for each function it appears in

In file included from ../video/out/gpu/shader_cache.c:9:
C:/MingW/media-autobuild_suite-master/local32/include/libavutil/mem.h:342:1: warning: 'alloc_size' attribute ignored on a function returning 'int' [-Wattributes]
  342 | av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
      | ^~~~~~~~~~~~~

Waf: Leaving directory `/build/mpv-git/build'
Build failed
 -> task in 'objects' failed with exit status 1 (run with -v to display more information)
build failed. Check C:/MingW/media-autobuild_suite-master/build/mpv-git/ab-suite.build.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/zhQi.zip
Copy and paste [logs.zip](https://0x0.st/zhQi.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```

**Comment by 1480c1** [comment]
It doesn't seem like you deleted `shaderc-git` folder
```none
shaderc git  ............................................. [Already built]
```

**Comment by donradzsk** [comment]
It keeps coming back after I delete it. Even after I deleted both shaderc-git and mpv-git.
```
ffmpeg git  ................................................. [Up-to-date]
├ Compiling mpv with Vapoursynth R48
├ mpv will need vapoursynth.dll and vsscript.dll to use vapoursynth filter!
mujs git  ................................................ [Already built]
vulkan-loader git  ....................................... [Already built]
  Running git clone for shaderc...
shaderc git  ............................................. [Already built]
SPIRV-Cross git  ......................................... [Already built]
libplacebo git  .......................................... [Already built]
  Running git clone for mpv...
┌ mpv git  ............................................ [Recently updated]
├ Running bootstrap...
├ Uninstalling python3-docutils... failed
├ Running configure...
├ Running build...
Likely error (tail of the failed operation logfile):
                 from C:/MingW/media-autobuild_suite-master/local32/include/libavutil/samplefmt.h:24,
                 from C:/MingW/media-autobuild_suite-master/local32/include/libavcodec/avcodec.h:31,
                 from ../audio/filter/af_lavcac3enc.c:28:
C:/MingW/media-autobuild_suite-master/local32/include/libavutil/mem.h:342:1: warning: 'alloc_size' attribute ignored on a function returning 'int' [-Wattributes]
  342 | av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
      | ^~~~~~~~~~~~~

Waf: Leaving directory `/build/mpv-git/build'
Build failed
 -> task in 'objects' failed with exit status 1 (run with -v to display more information)
build failed. Check C:/MingW/media-autobuild_suite-master/build/mpv-git/ab-suite.build.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/zhjS.zip
Copy and paste [logs.zip](https://0x0.st/zhjS.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```

**Comment by 1480c1** [comment]
How about deleting `local32/include/shaderc/shaderc.h`?

**Comment by donradzsk** [comment]
I had already deleted the entire build folder. Now I'm getting a tesseract error.
```
        Starting 64bit compilation of all tools
Imported default FFmpeg options from .bat
License: nonfree
Imported default mpv options from .bat

        Starting 64bit compilation of global tools
├ Running rustup_update...
├ Running set_default_toolchain...
  Running git update for sccache...
sccache git  ................................................ [Up-to-date]
  Running git update for ripgrep...
ripgrep git  ................................................ [Up-to-date]
  Running git update for jo...
jo git  ..................................................... [Up-to-date]
  Running git update for jq...
jq git  ..................................................... [Up-to-date]
  Running git update for dssim...
dssim git  .................................................. [Up-to-date]
  Running git update for libxml2...
libxml2 git  ................................................ [Up-to-date]
  Running git update for freetype2...
freetype2 git  .............................................. [Up-to-date]
  Running git update for fontconfig...
fontconfig git  ............................................. [Up-to-date]
  Running git update for harfbuzz...
harfbuzz git  ............................................... [Up-to-date]
  Running git update for fribidi...
fribidi git  ................................................ [Up-to-date]
  Running git update for libass...
libass git  ................................................. [Up-to-date]
gnutls 3.6.10  .............................................. [Up-to-date]
libidn2 2.2.0  .............................................. [Up-to-date]
libpsl 0.21.0  .............................................. [Up-to-date]
  Running git update for curl...
curl git  ................................................... [Up-to-date]
  Running git update for libtiff...
libtiff git  ................................................ [Up-to-date]
  Running git update for libwebp...
libwebp git  ................................................ [Up-to-date]
├ FFmpeg and related apps will depend on OpenCL.dll
  Running git update for OpenCL-Headers...
OpenCL-Headers git  ......................................... [Up-to-date]
  Running git update for leptonica...
leptonica git  .............................................. [Up-to-date]
  Running git update for tesseract...
┌ tesseract git  ...................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
C:/MingW/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/MingW/media-autobuild_suite-master/local64/lib/libcurl.a(libcurl_la-openssl.o):openssl.c:(.text+0x4d9d): undefined reference to `SSL_CTX_set_srp_password'
C:/MingW/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/MingW/media-autobuild_suite-master/local64/lib/libcurl.a(libcurl_la-openssl.o):openssl.c:(.text+0x5989): undefined reference to `OPENSSL_sk_pop'
C:/MingW/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/MingW/media-autobuild_suite-master/local64/lib/libcurl.a(libcurl_la-openssl.o):openssl.c:(.text+0x59c7): undefined reference to `OPENSSL_sk_num'
C:/MingW/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/MingW/media-autobuild_suite-master/local64/lib/libcurl.a(libcurl_la-openssl.o):openssl.c:(.text+0x59f9): undefined reference to `OPENSSL_sk_pop_free'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [Makefile:2369: tesseract.exe] Error 1
make[2]: Leaving directory '/build/tesseract-git/build-64bit'
make[1]: *** [Makefile:4352: all-recursive] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:1581: all] Error 2
make failed. Check C:/MingW/media-autobuild_suite-master/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/zh9E.zip
Copy and paste [logs.zip](https://0x0.st/zh9E.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```
What to delete if I don't need it?

**Comment by donradzsk** [comment]
How to fix either the tesseract error or this one, thanks.
`ERROR: cuda requested, but not all dependencies are satisfied: ffnvcodec`

**Comment by pat357** [comment]
Try to delete the folder **ffnvcodec-git** in /build and delete the ffnvcodec.pc in /local64/lib/pkgconfig
and remove the folder ffnvcodec in /local64/include.
```bash
rm -rf /build/ffnvcodec-git
rm -f /local64/lib/pkgconfig/ffnvcodec.pc
rm -rf /local64/include/ffnvcodec

**Comment by 1480c1** [comment]
@donradzsk, for the tesseract, there should be an already opened issue on tesseract with openssl symbols https://github.com/m-ab-s/media-autobuild_suite/issues/1440

For ffnvcodec, I would recommend you to try what @pat357 did, if not, open another issue on ffnvcodec specifically.

**Comment by donradzsk** [comment]
I'm still getting the same errors even after I deleted the ffnvcodec related stuff like pat357 suggested. I've since moved on to the ffmpeg-windows-build-helpers edition which is up and running.

**Comment by 1480c1** [comment]
Interesting, that's good that fwbh is working, still not sure why it's erroring out at ffnvcodec and can't reproduce on either of my machines and I don't have any logs
